modules/internal/String.chpl

```diff
@@ -452,7 +452,16 @@ module String {
     /*
       Iterates over the string Unicode character by Unicode character.
     */
-    iter uchars(): int(32) {
+    inline iter uchars(): int(32) {
+      for (codepoint, i, nbytes) in this.ucharsIndexLen() do
+        yield codepoint;
+    }
+
+    /*
+      Iterates over the string Unicode character by Unicode character,
+      and includes the byte index and byte length of each character.
+    */
+    inline iter ucharsIndexLen() {
```

Comment:
I'm not sure this is a great name for it. I'd consider making this an internal function for the moment.

Reply:
It also might be better to have the old iterator for performance reasons. Tuples are pretty fast but aren't free.

Reply:
Thanks.  I will make ucharsIndexLen an internal function and restore the old uchars iterator.
I'm not sure there is a good name for ucharsIndexLen.